packages/core/lib/triggers/TRPR0020.test.ts

```diff
@@ -0,0 +1,164 @@
+
+        const result = TRPR0020(generatedHearingOutcome)
+
+        expect(result).toEqual([
+          { code: triggerCode, offenceSequenceNumber: offenceSequenceNumber(generatedHearingOutcome) }
+        ])
+      })
+    })
+  })
+
+  offenceCodes.forEach((offenceCode) => {
+    it(`should not raise a trigger if result code equals 1085 (result code is not final and not excluded), is guilty and offence code matches ${offenceCode}`, () => {
+      const generatedHearingOutcome = generateMockAho(1085, offenceCode, CjsVerdict.Guilty)
+
+      const result = TRPR0020(generatedHearingOutcome)
+
+      expect(result).toEqual([])
+    })
+
+    it(`should not raise a trigger if result code equals 1085 (result code is not final and not excluded), is not guilty and offence code matches ${offenceCode}`, () => {
+      const generatedHearingOutcome = generateMockAho(1085, offenceCode, CjsVerdict.NotGuilty)
+
+      const result = TRPR0020(generatedHearingOutcome)
+
+      expect(result).toEqual([])
+    })
+  })
+
+  excludedResultCodes.forEach((resultCode) => {
+    offenceCodes.forEach((offenceCode) => {
+      it(`should not raise a trigger if result code equals ${resultCode} (result code is excluded), is guilty and offence code matches ${offenceCode}`, () => {
+        const generatedHearingOutcome = generateMockAho(resultCode, offenceCode, CjsVerdict.Guilty)
+
+        const result = TRPR0020(generatedHearingOutcome)
+
+        expect(result).toEqual([])
+      })
+
+      it(`should not raise a trigger if result code equals ${resultCode} (result code is excluded), is not guilty and offence code matches ${offenceCode}`, () => {
+        const generatedHearingOutcome = generateMockAho(resultCode, offenceCode, CjsVerdict.NotGuilty)
+
+        const result = TRPR0020(generatedHearingOutcome)
+
+        expect(result).toEqual([])
+      })
+    })
+  })
+
+  resultCodes.forEach((resultCode) => {
+    it(`should raise a trigger if result code equals ${resultCode} (result code is final), is guilty and offence code is not from the list`, () => {
+      const generatedHearingOutcome = generateMockAho(resultCode, "XXXXXXX", CjsVerdict.Guilty)
+
+      const result = TRPR0020(generatedHearingOutcome)
+
+      expect(result).toEqual([
+        { code: triggerCode, offenceSequenceNumber: offenceSequenceNumber(generatedHearingOutcome) }
+      ])
+    })
+
+    it(`should raise a trigger if result code equals ${resultCode} (result code is final), is not guilty and offence code offence code is not from the list`, () => {
+      const generatedHearingOutcome = generateMockAho(resultCode, "XXXXXXX", CjsVerdict.NotGuilty)
+
+      const result = TRPR0020(generatedHearingOutcome)
+
+      expect(result).toEqual([
+        { code: triggerCode, offenceSequenceNumber: offenceSequenceNumber(generatedHearingOutcome) }
+      ])
+    })
+  })
+
+  excludedResultCodes.forEach((resultCode) => {
```

Comment:
It seems `excludedResultCodes.forEach()` is duplicated. Can we not use one `excludedResultCodes.forEach()`?